sycl/plugins/unified_runtime/ur/adapters/native_cpu/program.hpp

```diff
@@ -31,3 +31,8 @@ struct ur_program_handle_t_ {
 
   std::map<const char *, const unsigned char *, _compare> _kernels;
 };
+
+struct nativecpu_entry {
```

Comment:
This certainly deserved some comment about what it is an how it is used. But isn't it defined somewhere else already? If we can't use that master definition directly here, then this should at least be a a big note that we need to keep in sync.

Reply:
This struct is only used for kernel lookup inside `program.cpp` which is why its definition was "hidden" inside that file before this change. Only the OffloadWrapper creates instances of this struct, but it does that directly in LLVM IR which is why there is no C++ definition of that struct anywhere else.

Reply:
> Only the OffloadWrapper creates instances of this struct, but it does that directly in LLVM IR

Could you point where this is done so I could better understand why it cannot be shared?
Also, please do add a source code comment about this.

Reply:
The struct is defined in the offload wrapper [here](https://github.com/PietroGhg/llvm/blob/de4ca64f479ea627521c20cb6d1571a0cb2fc874/clang/tools/clang-offload-wrapper/ClangOffloadWrapper.cpp#L631), it is a pattern followed by the other structs used by PI e.g. the [offload_entry](https://github.com/PietroGhg/llvm/blob/de4ca64f479ea627521c20cb6d1571a0cb2fc874/clang/tools/clang-offload-wrapper/ClangOffloadWrapper.cpp#L390) struct, the definition can't be directly shared in a straight forward way. I had the comment in the offload wrapper but not in the Plug-in code, I'll add the comment here since as you say it is much needed, thank you

Reply:
Then maybe move it to pi.h around here: https://github.com/intel/llvm/blob/a8cc71eb55171cef613d8e1077870fb0ea441b14/sycl/include/sycl/detail/pi.h#L953
?

Reply:
While I see the benefit of having all the structs needed by the offload-wrapper defined in the same file, maybe it would be better to keep the `nativecpu_entry` `struct defined in the Native CPU adapter, without exposing it to all the PI plug-ins. E.g. if another contributor makes a change to that struct we (Native CPU) maintainers would not get notified if we move it to `pi.h`, I think

Reply:
I am OK to keep it where it is, just add a comment

Reply:
I added the comment, thank you